src/app/pages/project/project.component.ts

```diff
@@ -36,45 +39,100 @@ export class ProjectComponent {
     ngOnInit() {
 
         this.route.paramMap.subscribe((params: ParamMap) => {
-            const activeReportId = params.get('id');
+            this.activeProjectId = params.get('id');
 
             // get current report based on param ID
             if (params.has('id') && this.projectService.projects) {
                 this.projectList = this.projectService.projects;
-                this.activeProject = this.projectList.length > 0 && this.projectService.projects.find((el) => el.id === activeReportId)
-                this.initChartData(this.activeProject);
+                this.activeProject = this.projectList.length > 0 && this.projectService.projects.find((el) => el.id === this.activeProjectId)
+                this.activeProjectRepo = this.activeProject.projectRepos.find(repo => repo.isDefault) || this.activeProject.projectRepos[0];
```

Comment:
Extract this right side of this expression into a method which returns the active repo and use it on the else branch as well.